magefile.go

```diff
@@ -8,31 +8,154 @@
 package main
 
 import (
+	"context"
+	"fmt"
+	"os"
 	"path/filepath"
+	"runtime"
+	"strconv"
+	"strings"
 
-	//mage:import
 	"github.com/elastic/elastic-agent-libs/dev-tools/mage"
+	devtools "github.com/elastic/elastic-agent-shipper/dev-tools/common"
+	"github.com/elastic/elastic-agent-shipper/tools"
+
+	//mage:import
+
 	"github.com/elastic/elastic-agent-libs/dev-tools/mage/gotool"
 	"github.com/magefile/mage/mg"
 	"github.com/magefile/mage/sh"
 )
 
-var (
-	goLicenserRepo = "github.com/elastic/go-licenser@v0.4.1"
+const (
+	GoreleaserRepo = "github.com/goreleaser/goreleaser@v1.6.3"
 )
 
 // Aliases are shortcuts to long target names.
 // nolint: deadcode // it's used by `mage`.
 var Aliases = map[string]interface{}{
-	"llc":  mage.Linter.LastChange,
-	"lint": mage.Linter.All,
+	"build":           Build.Binary,
+	"unitTest":        Test.Unit,
+	"integrationTest": Test.Integration,
+}
+
+// BUILD
+
+// Build contains targets related to linting the Go code
+type Build mg.Namespace
+
+// All builds binaries for the all  os/arch.
+func (Build) All() {
+	os.Setenv("PLATFORM", "all")
+	mg.Deps(Build.Binary)
+}
+
+// Clean removes the build directory.
+func (Build) Clean(test string) {
+	os.RemoveAll("build") // nolint:errcheck //not required
+}
+
+// CheckBinaries checks if the binaries are generated (for now).
+func (Build) CheckBinaries() error {
+	path := filepath.Join("build", "binaries")
+	for _, platform := range devtools.PlatformFiles {
+		var execName = "elastic-agent-shipper"
+		if strings.Contains(platform, "windows") {
+			execName += ".exe"
+		}
+		binary := filepath.Join(path, fmt.Sprintf("%s-%s-%s", "elastic-agent-shipper", tools.DefaultBeatVersion, platform), execName)
+		if _, err := os.Stat(binary); os.IsNotExist(err) {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// InstallGoReleaser target installs goreleaser
+func InstallGoReleaser() error {
+	return gotool.Install(
+		gotool.Install.Package(GoreleaserRepo),
+	)
+}
+
+// Binary will create the project binaries found in /build/binaries (use `mage build`)
+// ENV PLATFORM = all, darwin, linux, windows, darwin/amd64, darwin/arm64, linux/386, linux/amd64, linux/arm64, windows/386, windows/amd64
+// ENV SNAPSHOT = true/false
```

Comment:
On the beats and [agent](https://github.com/elastic/elastic-agent/blob/main/dev-tools/mage/build.go#L54-L62) there is a `DEV` env var to keep the debug symbols, avoid inlining and optimizations. It'd be good to have it here as well.

Reply:
good catch, added the necessary flags